### PR TITLE
resource_workspace_conf Delete: only reset enable and enforce properties

### PR DESCRIPTION
**Pull request by abij**
Make the code aligned with the documentation: https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/workspace_conf#argument-reference

PR for Issue: #1802 (remove a workspace with a non-enable/enforce setting)

### CODE REVIEW

File: workspace/resource_workspace_conf.go [hunk 1]
```diff
@@ -104,6 +104,11 @@ func ResourceWorkspaceConf() *schema.Resource {
 		Delete: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
 			config := d.Get("custom_config").(map[string]any)
 			for k, v := range config {
+				if !strings.HasPrefix(k, "enable") && !strings.HasPrefix(k, "enforce") {
```
Comment (alexott) [inline workspace/resource_workspace_conf.go:107]: Really it's not ideal situation as the workspace isn't returned to the original state. One approach would be to collect original values on create/update, and then reset to that values on delete.

Comment (abij) [reply workspace/resource_workspace_conf.go:107]: I get your point, for example the `projectsAllowList` (allowed git url's) is something you would like to reset (`""`), with the companion setting: `projectsAllowListPermissions
: "ALL_REPOS"` (default). 

Is there a way how to fetch or store the default values?

Comment (alexott) [reply workspace/resource_workspace_conf.go:107]: yes. something similar. 

You can retrieve original values when creating/updating a resource and store original values in a computed property (for example as JSON string, or something like)

Comment (Nestor10) [reply workspace/resource_workspace_conf.go:107]: a simpler solution could be to do nothing. Delete just removes the resource from backend / state. (This follows others work arounds now) The logic here would be that we are asking for the lack of management over the resource. As we know, we can't actually delete the workspace conf. All workspaces start with a configuration, even if its default. To get a default configuration, a simple inclusion of a workspace_conf with default settings. That to me IS a configuration. Its all grey but i fall into the camp that https://github.com/databricks/terraform-provider-databricks/blob/f29e655c5e71d00f09044dce323e16b6ab463b7b/workspace/resource_workspace_conf.go#L104
is asking for no more management not default setup.

**Comment by nkvuong** [comment]
Resolved by #3737

**Pull request closed without merging**